src/realm/query_expression.hpp

```diff
@@ -469,12 +475,13 @@ Query create(L left, const Subexpr2<R>& right)
     // TODO: recognize size operator expressions
     // auto size_operator = dynamic_cast<const SizeOperator<Size<StringData>, Subexpr>*>(&right);
 
-    if (column && ((std::numeric_limits<L>::is_integer && std::numeric_limits<R>::is_integer) ||
-                   (std::is_same<L, double>::value && std::is_same<R, double>::value) ||
-                   (std::is_same<L, float>::value && std::is_same<R, float>::value) ||
-                   (std::is_same<L, Timestamp>::value && std::is_same<R, Timestamp>::value) ||
-                   (std::is_same<L, StringData>::value && std::is_same<R, StringData>::value) ||
-                   (std::is_same<L, BinaryData>::value && std::is_same<R, BinaryData>::value)) &&
+    if (column &&
+        ((std::numeric_limits<L>::is_integer && std::numeric_limits<R>::is_integer) ||
+         (std::is_same<L, double>::value && std::is_same<R, double>::value) ||
```

Comment:
Should this support Decimal128?

Reply:
We don't yet have any optimizations for Decimal128 or ObjectID queries. At some point in the future, we will want to add a specialized `Decimal128Node` and `ObjectIDNode` which will speed things up significantly for queries involving those types. In the meantime, we use the default generic-but-slower query expression implementation.